### PR TITLE
Add MoEngage to React Native app

### PR DESCRIPTION
**Pull request by ishachopra0401**
## Description of the change

Added a tab for steps to add MoEngage to React Native app

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by 1abhishekpandey — APPROVED** [review]

File: destinations/marketing/moengage.mdx [hunk 1]
```diff
@@ -197,7 +197,22 @@ yarn add @rudderstack/rudder-integration-moengage-react-native
...
+const config = {
+  dataPlaneUrl: <DATA_PLANE_URL> ,
+  trackAppLifecycleEvents: true,
+  withFactories: [moengage],
+}
+rudderClient.setup( <WRITE_KEY> , config)
+```
+</span>
+</li>
+<li>Add the following under the dependencies section of the <code class="inline-code">app/build.gradle</code> (Module: Project) file in the Android folder of your React Native project:
```
Comment (desusai7) [inline destinations/marketing/moengage.mdx:215]:
```suggestion
<li>Add the following under the dependencies section of the <code class="inline-code">android/app/build.gradle</code> (Module: app) file in the Android folder of your React Native project:
```

File: destinations/marketing/moengage.mdx [hunk 2]
```diff
@@ -218,7 +233,7 @@ implementation 'com.google.code.gson:gson:2.8.6'
 ```
 </span>
 </li>
-<li>Open your <code class="inline-code">app/build.gradle</code> (Module: Project) file and add the following:
+<li>In the <code class="inline-code">app/build.gradle</code> (Module: Project) file, add the following:
```
Comment (desusai7) [inline destinations/marketing/moengage.mdx:236]:
```suggestion
<li>In the <code class="inline-code">android/build.gradle</code> (Module: Project) file, add the following:
```

File: destinations/marketing/moengage.mdx [hunk 3]
```diff
@@ -237,7 +252,7 @@ apply plugin: 'com.google.gms.google-services'
 ```
 </span>
 </li>
- <li>Open your <code class="inline-code">build.gradle</code> (Module: Project) file and add the following in <code class="inline-code">buildscript</code> dependencies:
+ <li>Add the following in <code class="inline-code">buildscript</code> dependencies in <code class="inline-code">build.gradle</code> (Module: Project) file:
```
Comment (desusai7) [inline destinations/marketing/moengage.mdx:255]:
```suggestion
 <li>Add the following in <code class="inline-code">buildscript</code> dependencies in <code class="inline-code">android/build.gradle</code> (Module: Project) file:
```

File: destinations/marketing/moengage.mdx [hunk 4]
```diff
@@ -237,58 +233,53 @@ implementation 'com.google.code.gson:gson:2.8.6'
...
-// For Push Notification
-apply plugin: 'com.google.gms.google-services'
+repositories {
+  mavenCentral()
+}
 ```
 </span>
 </li>
-<li>Open your <code class="inline-code">build.gradle</code> (Module: Project) file and add the following in <code class="inline-code">buildscript</code> dependencies
+ <li>Also, add the below plugin after <code class="inline-code">buildscript</code>:
```
Comment (desusai7) [inline destinations/marketing/moengage.mdx:246]:
```suggestion
<li>Also, add the below plugin to the top of the <code class="inline-code">android/app/build.gradle</code> (Module: app) :
```

**Comment by desusai7** [comment]
Can we also move point 5 -> 6 in React Native Device Mode Setup ?

File: destinations/marketing/moengage.mdx [hunk 5]
```diff
@@ -237,58 +233,53 @@ implementation 'com.google.code.gson:gson:2.8.6'
...
+buildscript {
+  dependencies {
+    classpath 'com.google.gms:google-services:4.3.4'
+  }
+}
 ```
 </span>
 </li>
-<li>Also, Initialize the MoEngage SDK in the Application class's onCreate() method as following:
+<li>Initialize the Rudder SDK in the <code class="inline-code">Application</code> class's <code class="inline-code">onCreate()</code> method as following:
```
Comment (desusai7) [inline destinations/marketing/moengage.mdx:268]:
```suggestion
<li>Initialize the MoEngage SDK in the <code class="inline-code">Application</code> class's <code class="inline-code">onCreate()</code> method as following:
```

**Review by desusai7 — APPROVED** [review]

File: destinations/marketing/moengage.mdx [hunk 6]
```diff
@@ -160,69 +180,45 @@ val moEngage = MoEngage.Builder(this, "XXXXXXXXXXX")
...
 <span>
 
-```groovy
-repositories {
-    mavenCentral()
+```javascript
+import rudderClient from "@rudderstack/rudder-sdk-react-native"
+import moengage from "@rudderstack/rudder-integration-moengage-react-native"
+const config = {
+  dataPlaneUrl: <DATA_PLANE_URL> ,
```
Comment (pallabmaiti) [inline destinations/marketing/moengage.mdx:207]: Fix the `,` after DATA_PLANE_URL

Comment (1abhishekpandey) [inline destinations/marketing/moengage.mdx:207]:
```suggestion
  dataPlaneUrl: DATA_PLANE_URL,
```

File: destinations/marketing/moengage.mdx [hunk 7]
```diff
@@ -59,48 +59,60 @@ By turning on the native SDK switch under Android SDK Settings on the RudderStac
...
+  <TabList>
+    <Tab>iOS</Tab>
+    <Tab>Android</Tab>
+    <Tab>React Native</Tab>
+  </TabList>
+    <TabPanels>
+      <TabPanel>
+        Follow these steps to add MoEngage to your iOS project:
+<ol>
+<li>Go your <code class="inline-code">Podfile</code> and add the <code class="inline-code">Rudder-Moengage</code> extension:
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:72]:
```suggestion
<li>Go to your <code class="inline-code">Podfile</code> and add the <code class="inline-code">Rudder-Moengage</code> extension:
```

File: destinations/marketing/moengage.mdx [hunk 8]
```diff
@@ -59,48 +59,60 @@ By turning on the native SDK switch under Android SDK Settings on the RudderStac
...
+<span>
 
 ```ruby
 pod 'Rudder-Moengage'
 ```
-
-- After adding the dependency followed by `pod install` , you can add the imports to your `AppDelegate.m` file as shown:
+</span>
+</li>
+<li>After adding the dependency followed by <code class="inline-code">pod install</code> , you can add the imports to your <code class="inline-code">AppDelegate.m</code> file as shown:
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:80]:
```suggestion
<li>After adding the dependency followed by <code class="inline-code">pod install</code>, you can add the imports to your <code class="inline-code">AppDelegate.m</code> file, as shown:
```

File: destinations/marketing/moengage.mdx [hunk 9]
```diff
@@ -59,48 +59,60 @@ By turning on the native SDK switch under Android SDK Settings on the RudderStac
...
+<span>
 
 ```objectivec
 #import "RudderMoengageFactory.h"
 ```
-
-- Finally, change the initialization of your `RudderClient` as shown:
+</span>
+</li>
+<li>Finally, change the initialization of your <code class="inline-code">RudderClient</code> as shown:
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:88]:
```suggestion
<li>Finally, change the initialization of your <code class="inline-code">RudderClient</code>, as shown:
```

File: destinations/marketing/moengage.mdx [hunk 10]
```diff
@@ -59,48 +59,60 @@ By turning on the native SDK switch under Android SDK Settings on the RudderStac
...
-
-To add MoEngage to your Android Project and enable its full functionality like push notifications, please follow these steps :
-
-- Open your `app/build.gradle` \(Module: app\) file, and add the following:
+</span>
+</li>
+</ol>
+      </TabPanel>
+      <TabPanel>
+Follow these steps to add MoEngage to your Android project and enable its full functionality, like push notifications:
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:102]:
```suggestion
Follow these steps to add MoEngage to your Android project:
```

File: destinations/marketing/moengage.mdx [hunk 11]
```diff
@@ -59,48 +59,60 @@ By turning on the native SDK switch under Android SDK Settings on the RudderStac
...
+<ol>
+<li>Open your <code class="inline-code">app/build.gradle</code> (Module: app) file, and add the following:
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:104]:
```suggestion
<li>Open your <code class="inline-code">app/build.gradle</code> (Module: app) file and add the following:
```

File: destinations/marketing/moengage.mdx [hunk 12]
```diff
@@ -118,15 +130,19 @@ implementation 'com.google.firebase:firebase-messaging:21.0.0'
...
 
 ```groovy
 // For Push Notification
 apply plugin: 'com.google.gms.google-services'
 ```
-
-- Open your `build.gradle` \(Module: Project\) file and add the following in `buildscript` dependencies
+</span>
+</li>
+<li>Open your <code class="inline-code">build.gradle</code> (Module: Project) file and add the following in <code class="inline-code">buildscript</code> dependencies:
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:144]:
```suggestion
<li>Open your <code class="inline-code">build.gradle</code> (Module: Project) file and add the following in the <code class="inline-code">buildscript</code> dependency:
```

File: destinations/marketing/moengage.mdx [hunk 13]
```diff
@@ -136,10 +152,12 @@ buildscript {
     }
 }
 ```
+</span>
+</li>
+<li>Initialize the Rudder SDK in the <code class="inline-code">Application</code> class's <code class="inline-code">onCreate()</code> method as following:
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:157]:
```suggestion
<li>Initialize the RudderStack SDK in the <code class="inline-code">Application</code> class's <code class="inline-code">onCreate()</code> method, as shown:
```

File: destinations/marketing/moengage.mdx [hunk 14]
```diff
@@ -150,8 +168,10 @@ val rudderClient = RudderClient.getInstance(
         .build()
 )
 ```
-
-- Also, Initialize the MoEngage SDK in the Application class's onCreate\(\) method as following:
+</span>
+</li>
+<li>Initialize the MoEngage SDK in the <code class="inline-code">Application</code> class's <code class="inline-code">onCreate()</code> method as following:
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:173]:
```suggestion
<li>Initialize the MoEngage SDK in the <code class="inline-code">Application</code> class's <code class="inline-code">onCreate()</code> method:
```

File: destinations/marketing/moengage.mdx [hunk 15]
```diff
@@ -237,58 +233,53 @@ implementation 'com.google.code.gson:gson:2.8.6'
...
-// For Push Notification
-apply plugin: 'com.google.gms.google-services'
+repositories {
+  mavenCentral()
+}
 ```
 </span>
 </li>
-<li>Open your <code class="inline-code">build.gradle</code> (Module: Project) file and add the following in <code class="inline-code">buildscript</code> dependencies
+ <li>Add the following in <code class="inline-code">buildscript</code> dependencies in <code class="inline-code">android/build.gradle</code> (Module: Project) file:
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:246]: should it be `dependency` or `dependencies`? Looks like a single dependency to me.

Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:246]:
```suggestion
 <li>Add the following in the <code class="inline-code">buildscript</code> dependencies in <code class="inline-code">android/build.gradle</code> (Module: Project) file:
```

File: destinations/marketing/moengage.mdx [hunk 16]
```diff
@@ -237,58 +233,53 @@ implementation 'com.google.code.gson:gson:2.8.6'
...
-    }
+  dependencies {
+    classpath 'com.google.gms:google-services:4.3.4'
+  }
 }
 ```
 </span>
 </li>
-<li>Initialize the Rudder SDK in the <code class="inline-code">Application</code> class's <code class="inline-code">onCreate()</code> method as following:
+ <li>Also, add the below plugin to the top of the <code class="inline-code">android/app/build.gradle</code> (Module: app):
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:259]:
```suggestion
 <li>Also, add the following plugin at the top of the <code class="inline-code">android/app/build.gradle</code> (Module: app):
```

File: destinations/marketing/moengage.mdx [hunk 17]
```diff
@@ -237,58 +233,53 @@ implementation 'com.google.code.gson:gson:2.8.6'
...
-        .withFactory(MoengageIntegrationFactory.FACTORY)
-        .build()
-)
+// For Push Notification
+apply plugin: 'com.google.gms.google-services'
 ```
 </span>
 </li>
-<li>Also, Initialize the MoEngage SDK in the Application class's onCreate() method as following:
+<li>Initialize the MoEngage SDK in the <code class="inline-code">Application</code> class's <code class="inline-code">onCreate()</code> method as following:
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:268]:
```suggestion
<li>Initialize the MoEngage SDK in the <code class="inline-code">Application</code> class' <code class="inline-code">onCreate()</code> method, as shown:
```

File: destinations/marketing/moengage.mdx [hunk 18]
```diff
@@ -813,10 +805,15 @@ For more details please follow the <a href="https://docs.moengage.com/docs/setti
 </li>
 </ul>
       </TabPanel>
+      <TabPanel>
+        <ul>
+       <li>Open <code class="inline-code">android</code> folder of your React Native app and do follow all the steps mentioned in <code class="inline-code">Android</code> tab of <a href="#configuring-push-notifications">Configuring Push Notifications</a></li>
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:810]:
```suggestion
       <li>Open the <code class="inline-code">android</code> folder of your React Native app and do follow all the steps mentioned in <code class="inline-code">Android</code> tab of <a href="#configuring-push-notifications">Configuring Push Notifications</a></li>
```

File: destinations/marketing/moengage.mdx [hunk 19]
```diff
@@ -813,10 +805,15 @@ For more details please follow the <a href="https://docs.moengage.com/docs/setti
...
+<li>Open <code class="inline-code">ios</code> folder of your React Native app and do follow all the steps mentioned in <code class="inline-code">iOS</code> tab of <a href="#configuring-push-notifications">Configuring Push Notifications</a></li>
```
Comment (ameypv-rudder) [inline destinations/marketing/moengage.mdx:811]:
```suggestion
<li>Open the <code class="inline-code">ios</code> folder of your React Native app and follow all the steps listed in the <code class="inline-code">iOS</code> tab of the <a href="#configuring-push-notifications">Configuring Push Notifications</a> section. </li>
```

**Review by ameypv-rudder — APPROVED** [review]

**Pull request merged**